components/TxConfirmationDialog.qml

```diff
@@ -0,0 +1,482 @@
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; secondary: false }
+            PropertyChanges { target: confirmButton; visible: false }
+            StateChangeScript { script: backButton.forceActiveFocus() }
+        }, State {
+            // creating or sending transaction, show tx details but don't show any button
+            name: "bottomText";
+            when: errorText.text =="" && bottomText.text !==""
+            PropertyChanges { target: txAmountText; visible: appWindow.transactionAmount !== "(all)" }
+            PropertyChanges { target: txAmountBusyIndicator; visible: !txAmountText.visible }
+            PropertyChanges { target: txFiatAmountText; visible: txAmountText.visible && persistentSettings.fiatPriceEnabled }
+            PropertyChanges { target: txDetails; visible: true }
+            PropertyChanges { target: bottomRow; visible: true }
+            PropertyChanges { target: buttons; visible: false }
+        }
+    ]
+
+    // same signals as Dialog has
+    signal accepted()
+    signal rejected()
+    signal closeCallback();
+
+    // background
+    MoneroEffects.GradientBackground {
+        anchors.fill: parent
+        fallBackColor: MoneroComponents.Style.middlePanelBackgroundColor
+        initialStartColor: MoneroComponents.Style.middlePanelBackgroundGradientStart
+        initialStopColor: MoneroComponents.Style.middlePanelBackgroundGradientStop
+        blackColorStart: MoneroComponents.Style._b_middlePanelBackgroundGradientStart
+        blackColorStop: MoneroComponents.Style._b_middlePanelBackgroundGradientStop
+        whiteColorStart: MoneroComponents.Style._w_middlePanelBackgroundGradientStart
+        whiteColorStop: MoneroComponents.Style._w_middlePanelBackgroundGradientStop
+        start: Qt.point(0, 0)
+        end: Qt.point(height, width)
+    }
+
+    // Make window draggable
+    MouseArea {
+        anchors.fill: parent
+        property point lastMousePos: Qt.point(0, 0)
+        onPressed: { lastMousePos = Qt.point(mouseX, mouseY); }
+        onMouseXChanged: root.x += (mouseX - lastMousePos.x)
+        onMouseYChanged: root.y += (mouseY - lastMousePos.y)
+    }
+
+    function open() {
+        // Center
+        root.x = parent.width/2 - root.width/2
+        root.y = 100
+        root.z = 11
+        root.visible = true;
+        
+        //clean previous error message
+        errorText.text = "";        
+    }
+    
+    function close() {
+        root.visible = false;
+        closeCallback();
+    }
+
+    // TODO: implement without hardcoding sizes
+    width: 580
+    height: 400
+
+    ColumnLayout {
+        id: mainLayout
+        spacing: 10
+        anchors.fill: parent
+        anchors.margins: 25
+
+        RowLayout {
+            id: column
+            Layout.topMargin: 10
+            Layout.fillWidth: true
+            anchors.top: parent.top
```

Comment:
Quick note here. Mixing Layouts and anchors is undefined behaviour.

See here a list of warnings:

```
2020-02-16 20:23:54.926	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.927	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.927	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.928	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:163:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:189:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.929	W qrc:/components/TxConfirmationDialog.qml:200:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:211:9: QML BusyIndicator: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:219:9: QML QQuickText: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:230:9: QML GridLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:364:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.930	W qrc:/components/TxConfirmationDialog.qml:398:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:109:9: QML ColumnLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
2020-02-16 20:23:54.931	W qrc:/components/SuccessfulTxDialog.qml:167:9: QML RowLayout: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
```

Reply:
I'm using log level 4 and these warning messages are not being displayed. How do you turn this on?

Reply:
Which Qt version are you using? 5.9 might not display the warning. You can install newer Qt versions using the installer.